minetest/minetest
# Add `core.spawn_tree_on_vmanip`

### Goal of the PR
- Fixes #13788
- and fixes #2151

### How does it work
- Adds a`core.spawn_tree_on_vmanip(vmanip, pos, treedef)` function which works like `core.spawn_tree`, but spawns a L-system tree onto a VoxelManip object instead of the map.

### To do
- Ready for Review.
~~(Waiting for the feature freeze to be done.)~~

### How to test

- Start devtest
- Use the `testitems:tree_spawner_vmanip` item to spawn some trees
- Use `testitems:tree_spawner` to see that normal tree spawning still works
- Read the change in `lua_api.md`
